### PR TITLE
Add support for Taildrop (file sharing)

### PR DESCRIPTION
**Pull request by juanfont**
A very low-hanging fruit here. This PR just adds support for Taildrop (tailscale filesharing, https://tailscale.com/kb/1106/taildrop/)

### CODE REVIEW

**Comment by juanfont** [comment]
Implements #118

**Comment by qbit** [comment]
FWIW you can do HTTP stuff at a UNIX domain socket with `net/http`:


```
        socClient := http.Client{
                Transport: &http.Transport{
                        DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
                                return net.Dial("unix", "/tmp/sockpost.sock")
                        },
                },
        }

        var res *http.Response
        res, err := socClient.Post("http://unix/place/stuff", "application/octet-stream", fileReader)
        if err != nil {
                log.Fatal(err)
        }
```

This would eliminate the need for an external dependency on curl.

**Comment by juanfont** [comment]
> This would eliminate the need for an external dependency on curl.

I am afraid we can't. We execute `curl` _inside_ the container where we run Tailscale.

File: integration_test.go [hunk 1]
```diff
@@ -529,6 +540,110 @@ func (s *IntegrationTestSuite) TestSharedNodes() {
...
+			command := []string{"touch", fmt.Sprintf("/tmp/file_from_%s", hostname)}
+			_, err := executeCommand(
+				&tailscale,
+				command,
+				[]string{},
+			)
+			assert.Nil(s.T(), err)
+			for peername, ip := range ips {
+				s.T().Run(fmt.Sprintf("%s-%s", hostname, peername), func(t *testing.T) {
+					// We currently cant send files to so skip that
```
Comment (kradalby) [inline integration_test.go:560]: This comment sounds a bit unfinished

File: integration_test.go [hunk 2]
```diff
@@ -529,6 +540,110 @@ func (s *IntegrationTestSuite) TestSharedNodes() {
...
+						// Under normal circumstances, we should be able to send a file
+						// using `tailscale file cp` - but not in userspace networking mode
+						// So curl!
+						peerAPI, ok := apiURLs[ip]
+						assert.True(t, ok)
+
+						// We still have some issues with the test infrastructure, so
+						// lets run curl multiple times until it works.
+
+						attempts := 0
```
Comment (kradalby) [inline integration_test.go:572]: Lets add a `TODO(juanfont):` here to remove the retry loop when we get that sorted.

**Review by kradalby — APPROVED** [review]

**Comment by kradalby** [comment]
Looks good

**Pull request merged**